app/main.py

```diff
@@ -1 +1,36 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+
+    alive: list[Animal] = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name: str = name
+        self.health: int = max(min(health, 100), 0)
```

Comment:
based on the task I assume that 100 is not the max health, and it's hard to read, so it's better not to complicate things